code/modules/projectiles/guns/energy.dm

```diff
@@ -593,4 +548,348 @@
 	var/obj/screen/ammo/A = user.hud_used.ammo //The ammo HUD
 	A.update_hud(user)
 
+/obj/item/weapon/gun/energy/lasgun/lasrifle/update_item_state(mob/user) //Without this icon states for wielded guns won't show.
```

Comment:
```suggestion
/obj/item/weapon/gun/energy/lasgun/lasrifle/update_item_state(mob/user) //Without this override icon states for wielded guns won't show. because lasgun overrides and this has no charge icons
```